### DISCUSSION TITLE
RaspberryPi as RTU-TCP gateway for Solax X3HybridG4

### DISCUSSION THREAD

**Discussion by peterax** [wills106/homeassistant-solax-modbus#889] [Show and tell]
Put 2 of these [https://www.aliexpress.com/item/1005001580305871.html ](https://www.aliexpress.com/item/1005001580305871.html) in a RaspberryPi. One is connected to the inverters COM port and the other one to CT/Meter port.

Made a flow in NodeRED which does the gatewaying from TCP to RTU for Home Assistant and it works just fine!
The other port I hope to get working as a smart meter emulator. Just need to figure out what data to return to the inverter for each request...

**Comment by raenji-sk** [comment]
@peterax Would you care to elaborate and share more details of wiring and NoreRed flow code of your setup? Which inverter do you have?
